### PR TITLE
Show line/col for parsing exceptions in gen-s-parser

### PR DESCRIPTION
**Pull request by aheejin**
Not sure how to test it. Can we add a test that xfails in test/?
This is pretty straightforward anyway though.

### CODE REVIEW

File: test/unit/test_parsing_error.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from scripts.test.shared import WASM_OPT, run_process
+from utils import BinaryenTestCase
+import os
+
+
+class ParsingErrorTest(BinaryenTestCase):
+  def test_parsing_error_msg(self):
+    module = '''(module
```
Comment (tlively) [inline test/unit/test_parsing_error.py]: I've been formatting these to line up with the surrounding Python code so they don't stick out so much. For example,
```suggestion
    module = '''
    (module
     (func $foo
      (abc)
     )
    )
    '''
Of course this has the downside of changing the contents of the string, but I think that should be fine for this test. 

```

Comment (aheejin) [reply test/unit/test_parsing_error.py]: Oh this feature is cool! How can I suggest a change?

Comment (tlively) [reply test/unit/test_parsing_error.py]: The leftmost icon above the text box. Unfortunately I seem to have screwed this one up.

Comment (aheejin) [reply test/unit/test_parsing_error.py]: Oh but your comment got committed together 😂

File: test/unit/test_parsing_error.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+class ParsingErrorTest(BinaryenTestCase):
+  def test_parsing_error_msg(self):
+    module = '''
+    (module
+     (func $foo
+      (abc)
+     )
+    )
+    '''
+Of course this has the downside of changing the contents of the string, but I think that should be fine for this test.
```
Comment (tlively) [inline test/unit/test_parsing_error.py]:
```suggestion

```

File: test/unit/test_parsing_error.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    module = '''
+    (module
+     (func $foo
+      (abc)
+     )
+    )
+    '''
+Of course this has the downside of changing the contents of the string, but I think that should be fine for this test. 
+
+ (func $foo
```
Comment (tlively) [inline test/unit/test_parsing_error.py]:
```suggestion
```

File: test/unit/test_parsing_error.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  (abc)
```
Comment (tlively) [inline test/unit/test_parsing_error.py]:
```suggestion
```

File: test/unit/test_parsing_error.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  def test_parsing_error_msg(self):
+    module = '''
+    (module
+     (func $foo
+      (abc)
+     )
+    )
+    '''
+Of course this has the downside of changing the contents of the string, but I think that should be fine for this test. 
+
```
Comment (tlively) [inline test/unit/test_parsing_error.py]:
```suggestion
```

File: test/unit/test_parsing_error.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+ (func $foo
+  (abc)
+ )
```
Comment (tlively) [inline test/unit/test_parsing_error.py]:
```suggestion
```

File: test/unit/test_parsing_error.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+)
```
Comment (tlively) [inline test/unit/test_parsing_error.py]:
```suggestion
```

File: test/unit/test_parsing_error.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    '''
```
Comment (tlively) [inline test/unit/test_parsing_error.py]:
```suggestion
```

Comment (tlively) [reply test/unit/test_parsing_error.py]: what a mess

File: src/passes/Inlining.cpp [hunk 9]
```diff
@@ -77,20 +57,25 @@ struct FunctionInfo {
...
     }
     // if it has one use, then inlining it would likely reduce code size
     // since we are just moving code around, + optimizing, so worth it
     // if small enough that we are pretty sure its ok
-    if (calls == 1 && !usedGlobally && size <= CAREFUL_SIZE_LIMIT) {
+    // FIXME: move this check to be first in this function, since we should
+    // return true if oneCallerInlineMaxSize is bigger than
+    // flexibleInlineMaxSize (which it typically should be).
+    if (calls == 1 && !usedGlobally &&
+        size <= options.inlining.oneCallerInlineMaxSize) {
```
Comment (tlively) [inline src/passes/Inlining.cpp]: Looks like unrelated inlining changes snuck in?

Comment (aheejin) [reply src/passes/Inlining.cpp]: Fixed it
